core/src/main/scala/org/http4s/QueryParam.scala

```diff
@@ -195,4 +195,5 @@ object QueryParamDecoder {
       def decode(value: QueryParameterValue): ValidatedNel[ParseFailure, String] =
         value.value.validNel
     }
+  implicit lazy val uriQueryParamDecoder: QueryParamDecoder[Uri] = QueryParamDecoder[String].map(Uri.unsafeFromString)
```

Comment:
This will throw if the String is an invalid `Uri`.  We should have a function on `QueryParamDecoder` that works through a `ParseResult`, so we can use the safe conversion from `Uri`.

Reply:
What should be done for a Parse failure?